source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -361,6 +365,20 @@ absl::Status ProcessorState::handleTrailersResponse(const TrailersResponse& resp
   return absl::FailedPreconditionError("spurious message");
 }
 
+void ProcessorState::clearRouteCache(const CommonResponse& common_response) {
+  // Only clear the route cache if there is a mutation to the header and clearing is allowed.
+  if (filter_.config().disableClearRouteCache()) {
+    filter_.stats().clear_route_cache_disabled_.inc();
+    ENVOY_LOG(debug, "NOT clearing route cache, it is disabled in the config");
+  } else if (common_response.has_header_mutation()) {
+    ENVOY_LOG(debug, "clearing route cache");
+    filter_callbacks_->downstreamCallbacks()->clearRouteCache();
+  } else {
+    filter_.stats().clear_route_cache_ignored_.inc();
+    ENVOY_LOG(debug, "NOT clearing route cache, no header mutations detected");
```

Comment:
LGTM, but can I ask why we bundled the "avoid clearing route cache when no header mutations" into this PR?

IMHO it probably makes sense, although there might be valid reasons (albeit contrived) to use ext_proc to force a repick without this, e.g. if an earlier filter had set some headers and we're not having them take effect. As I said, a bit contrived and it's probably fine, just not sure why we needed it in this PR.

Reply:
I had included it in the scope of this pr earlier on because I had assumed it was part of the plan for the change, it was mentioned in one of the readiness documents in some notes. It seemed related enough to include. 
Though, it does make sense to split it up into another pr. What do you think?

Reply:
IMHO, if earlier filter mutate the header to affect route selection, technically that filter should clear route cache itself in place. Leaving it to other filters to clear seems not very robust behavior to me.

Yea, i don't have strong opinion. Either this PR or separate PR works for me.

Reply:
OK, I don't see any objections to combining, but please keep in mind for future PRs that ideally we have orthogonal PRs to support things like bisecting and change / risk management.